packages/components/src/base-control/types.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import type { HTMLAttributes, ReactNode } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type {
+	Wrapper,
+	StyledField,
+	StyledVisualLabel,
+} from './styles/base-control-styles';
+
+export type BaseControlProps = {
+	/**
+	 * Start opting into the new margin-free styles that will become the default in a future version.
+	 *
+	 * @default false
+	 */
+	__nextHasNoMarginBottom?: boolean;
+	/**
+	 * The HTML `id` of the element (passed in as a child to `BaseControl`) to which labels and help text are being generated.
+	 * This is necessary to accessibly associate the label with that element.
+	 */
+	id?: string;
+	/**
+	 * If this property is added, a help text will be generated using help property as the content.
+	 */
+	help?: ReactNode;
+	/**
+	 * If this property is added, a label will be generated using label property as the content.
+	 */
+	label?: ReactNode;
+	/**
+	 * If true, the label will only be visible to screen readers.
+	 *
+	 * @default false
+	 */
+	hideLabelFromVision?: boolean;
+	className?: HTMLAttributes< typeof Wrapper >[ 'className' ];
+	/**
+	 * The content to be displayed within the `BaseControl`.
+	 */
+	children?: HTMLAttributes< typeof StyledField >[ 'children' ];
```

Comment:
Curious to know why you didn't put `string` and `ReactNode` as the type of these two props?

Reply:
Haha, too much indirection? 😂 Part of it stems from the fact that I'm never confident about what the canonical types are for these built-in props. Like, does `className` accept `null`? Does `children` accept different things depending on element type? I wouldn't know without looking it up!

Do you have a preference? I prefer the indirection slightly, just for precision and ease of mind, but no strong opinion.

Reply:
> Like, does className accept null? 

If you asked me on the spot, I'd say "no" — in my mind it's either a `string`, or it's not specified (`undefined`)

>  Does children accept different things depending on element type?

Maybe? Although `children` is a standard React prop and its type is usually `React.ReactNode`

> Do you have a preference? I prefer the indirection slightly, just for precision and ease of mind, but no strong opinion.

I personally always used `className?: string` and `children?: React.ReactNode` (although with the introduction of `WordPressComponentsProps` we don't really specify `className` anymore). Having said that, I don't have a strong opinion either, we can go with whatever you think works best.

Reply:
Yeah... it's probably useful in some cases, but here I'm starting to think it's a bit overkill. Simplified again in https://github.com/WordPress/gutenberg/pull/39468/commits/8d13aedfe112136b07c0428c4f655a18a3bd4284